File: 2018/tests/Day11Test.hs
module Day11Test where

import           Day11
import           Test.Tasty.Hspec

spec_Day11 = do

  describe "Power level" $ do
    it "8 (3,5)" $ cellPower 8 (3, 5) `shouldBe` 4
    it "57 (122,79)" $ cellPower 57 (122, 79) `shouldBe` (-5)
    it "39 (217,196)" $ cellPower 39 (217, 196) `shouldBe` 0
    it "71 (101,153)" $ cellPower 71 (101, 153) `shouldBe` 4

  describe "Part 1" $ do
    it "example 1" $ part1 18 `shouldBe` (33, 45)
    it "example 2" $ part1 42 `shouldBe` (21, 61)
    it "my input" $ part1 7511 `shouldBe` (21, 22)

  -- TODO runs ~ 5 sec
  describe "Part 2" $ do
    -- it "example 1" $ part2 18 `shouldBe` (90, 269, 16)
    -- it "example 2" $ part2 42 `shouldBe` (232, 251, 12)
    it "my input" $ part2 7511 `shouldBe` (235, 288, 13)

File: 2021/test/day11_test.clj
(ns day11-test
  (:require
   [clojure.java.io :as io]
   [clojure.test :refer [deftest is testing]]
   [day11 :refer [part1 part2]]))

(def example
  "5483143223
2745854711
5264556173
6141336146
6357385478
4167524645
2176841721
6882881134
4846848554
5283751526")

(def my-input (slurp (io/resource "input11.txt")))

(deftest part1-test
  (testing "example"
    (is (= 1656 (part1 example))))
  (testing "my input"
    (is (= 1585 (part1 my-input)))))

(deftest part2-test
  (testing "example"
    (is (= 195 (part2 example))))
  (testing "my input"
    (is (= 382 (part2 my-input)))))
